client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+	timestamp := time.Now().String()
+	plain := map[string]string{
+		"secretType":  secret.SecretType,
+		"secretName":  secret.SecretName,
+		"description": secret.Description,
+		"version":     timestamp,
+	}
+	// var createdSecret *Secret
+	var createdRecord *pdsClient.Record
+	if secret.SecretType == "File" {
+		createdRecord, err = c.WriteFile(ctx, recordType, plain, secret.FileName)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		data := map[string]string{"secretValue": secret.SecretValue}
+		createdRecord, err = c.WriteRecord(ctx, data, recordType, plain)
+		if err != nil {
+			return nil, err
+		}
+	}
+	createdSecret := c.MakeSecretResponse(createdRecord, group.GroupID.String(), "")
+	err = c.ShareRecordWithGroup(ctx, recordType, group)
+	if err != nil {
+		return nil, err
+	}
+	return createdSecret, nil
+}
+
+// WriteFile encrypts the file with specified fileName and uploads it, creating a new record in E3DB
+func (c *ToznySDKV3) WriteFile(ctx context.Context, recordType string, plain map[string]string, fileName string) (*pdsClient.Record, error) {
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	ak, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return nil, err
+	}
+	// Encrypt the file
+	size, checksum, err := e3dbClients.EncryptFile(fileName, encryptedFileName, ak)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err := os.Remove(encryptedFileName)
+		if err != nil {
+			fmt.Println("CreateSecret: error removing encrypted")
+		}
+	}()
+	plain[SECRET_FILENAME_METADATA_KEY] = fileName
+	sizeKB := size / 1024
+	if sizeKB > 1 {
+		plain[SECRET_FILE_SIZE_METADATA_KEY] = fmt.Sprintf("%d", sizeKB)
+	} else {
+		plain[SECRET_FILE_SIZE_METADATA_KEY] = "< 1"
+	}
+	// Write the whole file
+	recordToWrite := storageClient.Record{
+		Metadata: storageClient.Meta{
+			Type:     recordType,
+			WriterID: uuid.MustParse(c.StorageClient.ClientID),
+			UserID:   uuid.MustParse(c.StorageClient.ClientID),
+			Plain:    plain,
+			FileMeta: &storageClient.FileMeta{
+				Size:        int64(size),
+				Checksum:    checksum,
+				Compression: "raw",
+			},
+		},
+	}
+	pendingFileURL, err := c.StorageClient.WriteFile(ctx, recordToWrite)
+	if err != nil {
+		return nil, err
+	}
+	uploadRequest := file.UploadRequest{
+		URL:               pendingFileURL.FileURL,
+		EncryptedFileName: encryptedFileName,
+		Checksum:          checksum,
+		Size:              size,
+	}
+	uploadResp, err := file.UploadFile(uploadRequest)
+	if err != nil || uploadResp != 0 {
+		return nil, err
+	}
+	// Register the file as being written
+	response, err := c.StorageClient.FileCommit(ctx, pendingFileURL.PendingFileID)
+	if err != nil {
+		return nil, err
+	}
+	// get the file from the record
+	fileSecret := &pdsClient.Record{
```

Comment:
confused as to the presence of secret in this variable name, suggest `fileRecord`